### PR TITLE
Manila generic driver config

### PR DESCRIPTION
**Pull request by toabctl**

### CODE REVIEW

**Comment by toabctl** [comment]
I need some feedback about the DHSS=true/false handling in crowbar. I think for now we only need to support DHSS=false but I wonder if we should create separate crowbar backends (i.e. generic-dhss-true and generic-dhss-false) for both cases. That has the advantage that not all parameters need to be required=false (which means we need to check mandatory paramters for each case (dhss true/false)). Opinions?

File: chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    share["generic"]["driver_handles_share_servers"] =
+      ta["share_defaults"]["generic"]["driver_handles_share_servers"]
+    share["generic"]["service_image_name"] =
+      ta["share_defaults"]["generic"]["service_image_name"]
+    share["generic"].delete("service_instance_name_or_id")
+    share["generic"].delete("service_net_name_or_ip")
+    share["generic"].delete("tenant_net_name_or_ip")
+  end
+  return a, d
+  end
```
Comment (mjura) [inline chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb]: Please fix hound alert

**Comment by mjura** [comment]
Please fix hound alerts

File: chef/data_bags/crowbar/template-manila.json [hunk 2]
```diff
@@ -57,9 +60,9 @@
           "backend_name": "default",
           "generic": {
             "service_instance_user": "root",
-            "service_instance_password": "",
-            "service_image_name": "manila-service-image",
-            "driver_handles_share_servers": true
+            "service_instance_name_or_id": "",
+            "service_net_name_or_ip": "",
+            "tenant_net_name_or_ip": ""
```
Comment (vuntz) [inline chef/data_bags/crowbar/template-manila.json]: Aren't you missing some required attributes here?

File: chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  # update shares
+  a["shares"].each do |share|
+    next if share["backend_driver"] != "generic"
+    share["generic"]["driver_handles_share_servers"] =
+      ta["share_defaults"]["generic"]["driver_handles_share_servers"]
+    share["generic"]["service_image_name"] =
+      ta["share_defaults"]["generic"]["service_image_name"]
+    share["generic"].delete("service_instance_name_or_id")
+    share["generic"].delete("service_net_name_or_ip")
+    share["generic"].delete("tenant_net_name_or_ip")
```
Comment (vuntz) [inline chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb]: You need to drop share_volume_fstype here.

File: chef/data_bags/crowbar/template-manila.json (hunk 2 above)
Comment (vuntz) [inline chef/data_bags/crowbar/template-manila.json]: Ah no, they're optional, sorry for the noise.

**Comment by vuntz** [comment]
Except for the migration issue, looks good to me.

**Comment by dirkmueller** [comment]
LGTM, +1

File: chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb (hunk 3 above)
Comment (toabctl) [inline chef/data_bags/crowbar/migrate/manila/012_fix_generic_driver_dhss_false.rb]: Done in https://github.com/crowbar/crowbar-openstack/commit/9db7faeb090c0270fde4e94c44ba603264d31a4d

**Comment by toabctl** [comment]
@dirkmueller addressed your commit comment in https://github.com/crowbar/crowbar-openstack/commit/9db7faeb090c0270fde4e94c44ba603264d31a4d

**Comment by vuntz** [comment]
+1

**Comment by toabctl** [comment]
Setting "do not merge yet" because we need https://github.com/SUSE-Cloud/automation/pull/742 to not kill our CI

**Comment by toabctl** [comment]
Just rebased.

**Comment by toabctl** [comment]
just rebased

File: crowbar_framework/config/locales/manila/en.yml [hunk 4]
```diff
@@ -57,3 +61,10 @@ en:
             manual:
               config: 'Options'
               driver: 'Driver'
+      validation:
+        generic:
+          service_instance_user: 'Service instance user can not be empty'
+          service_instance_name_or_id: 'Service instance Name/ID can not be empty'
+          service_net_name_or_ip: 'Service network name/IP can not be empty'
+          tenant_net_name_or_ip: 'Tenant network name/IP can not be empty'
+          password_or_private_key: 'Password or private key path must be given'
```
Comment (aplanas) [inline crowbar_framework/config/locales/manila/en.yml]: extra line?

**Comment by aplanas** [comment]
+1 for me

**Comment by toabctl** [comment]
3x +1 and gating passed. merging...

**Pull request merged**